addons/payment/views/payment_provider_views.xml

```diff
@@ -63,9 +65,9 @@
                         Please use the <em>Duplicate</em> action instead.
                     </div>
                     <group>
-                        <group name="payment_state">
-                            <field name="code" groups="base.group_no_one" attrs="{'readonly': [('id', '!=', False)], 'invisible': [('module_id', '!=', False), ('module_state', '!=', 'installed')]}"/>
-                            <field name="state" widget="radio" attrs="{'invisible': [('module_state', '=', 'uninstalled')]}"/>
+                        <group name="payment_state" attrs="{'invisible': [('module_state', '=', 'uninstalled')]}">
+                            <field name="code" groups="base.group_no_one" readonly="1"/>
```

Comment:
`code` needs to keep `'readonly': [('id', '!=', False)]` because it only makes the field readonly when the record is *created*. When it is *being created* (`id = False`), it should not be readonly.

Reply:
updated this also